#### Import Modules

In [5]:
import sys, os
import pandas as pd
import cohere
from dotenv import load_dotenv
from nltk.tokenize import word_tokenize
sys.path.append(os.path.abspath(os.path.join('../scripts')))
from read_write_util import ReadWriteUtil
from prompt_pipeline import PromptPipeline

In [3]:
reader = ReadWriteUtil()
API_KEY = os.getenv('API_KEY')
# print(API_KEY)
#Initialize reader and API_KEY

In [6]:
df = pd.read_csv('../data/news.csv')
df.head()
#Read data

Domain  \
0         rassegnastampa.news   
1                 twitter.com   
2             atpe-tchad.info   
3  badbluetech.bitnamiapp.com   
4           kryptogazette.com   

                                               Title  \
0  Boris Johnson using a taxpayer-funded jet for ...   
1  Stumbled across an interesting case, a woman f...   
2  Marché Résines dans les peintures et revêtemen...   
3        AI drives data analytics surge, study finds   
4  Triacetin Vertrieb Markt 2021: Globale Unterne...   

                                         Description  \
0  …often trigger a protest vote that can upset…t...   
1  Stumbled across an interesting case, a woman f...   
2  …COVID-19…COVID…COVID…COVID-19 et Post COVID…C...   
3  …hate raiders' linked to automated harassment ...   
4  …Abschnitten und Endanwendungen / Organisation...   

                                                Body  \
0  Boris Johnson using a taxpayer-funded jet for ...   
1  Stumbled across an interesting case, a woman f...   
2  Le rapport d’étude de marché Résines dans les ...   
3  How to drive the funnel through content market...   
4  Global Triacetin Vertrieb-Markt 2021 von Herst...   

                                                Link  \
0  https://rassegnastampa.news/boris-johnson-usin...   
1  http://twitter.com/CoruscaKhaya/status/1435585...   
2  http://atpe-tchad.info/2021/09/13/marche-resin...   
3  http://badbluetech.bitnamiapp.com/p.php?sid=21...   
4  https://kryptogazette.com/2021/09/08/triacetin...   

                    timestamp  Analyst_Average_Score  Analyst_Rank  \
0  2021-09-09T18:17:46.258006                    0.0             4   
1  2021-09-08T13:02:45.802298                    0.0             4   
2  2021-09-13T07:32:46.244403                    0.0             4   
3  2021-09-11T00:17:45.962605                    0.0             4   
4  2021-09-08T12:47:46.078369                    0.0             4   

   Reference_Final_Score  
0                   1.96  
1                  12.00  
2                   0.05  
3                   6.10  
4                   0.13

`Lets check for lengths of the body`

In [11]:
lengths = df['Body'].apply(lambda x: len(x))
lengths


0    13712
1      267
2     8273
3    22098
4     9554
5     1899
6     1729
7     3061
8     4428
9     5257
Name: Body, dtype: int64

`We can see that the lowest number of charachters we have is 267 most are above 2048 which is the highest number of tokens accepted by cohere, so we have to find a new approach to preprocess this data`

`We will try to understand the context of the body`